kbase-extension/static/kbase/js/widgets/function_output/kbaseConditionSetViewer.js

```diff
@@ -73,29 +75,28 @@ define (
                 function(ret) {
                     console.log(ret);
                     var cs = ret.data[0].data;
-                    for (var index = 0; index < cs.factors.length; ++index) {
-                        // init with optional values
-                        var row = {"unit": "", "unit_ont_ref": "", "unit_ont_id": ""};
-                        for (var key in cs.conditions){
-                            row[key] = cs.conditions[key][index]
+                    var rows = [];
+                    var cols = [{title: "Sample ID"}];
+                    cs.factors.forEach(function(factor){
+                        cols.push({title: factor.factor});
+                    });
+                    for (var _id in cs.conditions) {
```

Comment:
In this block, if that's how `rows` is getting initialized, you could use the `map` function instead. It looks like you're converting from key-value pairs to an array of arrays (e.g. from {'foo': 'bar'} to [['foo', 'bar']]). This'd be something like:

```
rows = Object.keys(cs.conditions).map(function(_id) {
    return [_id, cs.conditions[_id]];
});
```

Reply:
That's not quite what I need to do but there is still a nicer way to write this with map that I switched to